src/client.rs

```diff
@@ -32,9 +92,16 @@ impl Client {
         }
     }
 
-    pub fn get(url: &str) -> Client {
-        Client::new()
-            .with_endpoint(url.parse().unwrap())
+    pub fn get(url: &str) -> Result<Client, Error> {
+        let mut client = Client::new();
+        let url = Uri::new(url).map_err(Error::url_parsing)?;
+
+        let (endpoint, options) = decompose(url)?;
+
+        client.set_endpoint(endpoint);
+        client.msg.options = options;
```

Comment:
It's not super important within the library, but in general I was trying to avoid directly accessing the fields of the structs because I want to make sure everything I needed to do in the code here was also possible for consumers of the library to do.

Having said this particular instance is probably fine because I haven't settled on what the msg interfaces will look like. (And they probably won't exist directly on the client itself, but rather on Endpoint or a 'request' type that does not exist at the moment.) Just thought I might give some of my thoughts while I'm thinking of it.